include/boost/json/conversion.hpp

```diff
@@ -0,0 +1,293 @@
+    (std::tuple_size<T>::value > 0) >;
+
+template<class T>
+using has_unique_keys = has_positive_tuple_size<decltype(
+    std::declval<T&>().emplace(
+        std::declval<value_type<T>>()))>;
+
+template<class T>
+struct is_value_type_pair_helper : std::false_type
+{ };
+template<class T1, class T2>
+struct is_value_type_pair_helper<std::pair<T1, T2>> : std::true_type
+{ };
+template<class T>
+using is_value_type_pair = is_value_type_pair_helper<value_type<T>>;
+
+} // namespace detail
+
+/** Determine if `T` can be treated like a string during conversions.
+
+    Provides the member constant `value` that is equal to `true`, if `T` is
+    convertible to @ref string_view. Otherwise, `value` is equal to `false`.
+    <br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like strings. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_string_like<your::string> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+    @par Types satisfying the trait
+
+    @ref string,
+    @ref string_view,
+    <a href="https://en.cppreference.com/w/cpp/string/basic_string"><tt>std::string</tt></a>,
+    <a href="https://en.cppreference.com/w/cpp/string/basic_string_view"><tt>std::string_view</tt></a>.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_string_like
+#ifndef BOOST_JSON_DOCS
+    : std::is_convertible<T, string_view>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a sequence during conversions.
+
+    Given `t`, a glvalue of type `T`, if
+
+    @li given `It`, the type denoted by `decltype(std::begin(t))`,
+        <tt>std::iterator_traits<It>::iterator_category</tt> is well-formed and
+        denotes a type; and
+
+    @li `decltype(std::end(t))` also denotes the type `It`;
+
+    then the trait provides the member constant `value` that is equal to
+    `true`. Otherwise, `value` is equal to `false`.<br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like sequences. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_sequence_like<your::container> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+
+    @par Types satisfying the trait
+
+    Any <a href="https://en.cppreference.com/w/cpp/named_req/SequenceContainer"><em>SequenceContainer</em></a>,
+    array types.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_sequence_like
+#ifndef BOOST_JSON_DOCS
+    : mp11::mp_all<
+        mp11::mp_valid_and_true<detail::are_begin_and_end_same, T>,
+        mp11::mp_valid<detail::begin_iterator_category, T>>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a 1-to-1 mapping during
```

Comment:
Metafunction briefs should start with "Metafunction"

Reply:
These aren't metafunctions, they are type traits.

Reply:
it isn't a function returning a value?